src/api/MetadataQuery.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @flow
+ * @file Helper for the box metadata query API
+ * @author Box
+ */
+
+import Base from './Base';
+import { CACHE_PREFIX_METADATA, ERROR_CODE_METADATA_QUERY } from '../constants';
+
+class MetadataQuery extends Base {
+    /**
+     * @property {string}
+     */
+    key: string;
+
+    /**
+     * @property {Function}
+     */
+    successCallback: Function;
+
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * Creates a key for the metadata cache
+     *
+     * @param {string} id - metadata template
+     * @return {string} key
+     */
+    getCacheKey(id: string): string {
+        return `${CACHE_PREFIX_METADATA}query_${id}`;
```

Comment:
Should we just keep them separate and create a new constant for the metadata query cache prefix?